File: scripts/fetch-logs.ts
import { Client } from 'ssh2';

const conn = new Client();

const config = {
  host: '37.60.249.75',
  port: 22,
  username: 'root',
  password: 'Aliwdansaa23',
  readyTimeout: 99999
};

console.log('Connecting to Contabo to fetch logs...');

conn.on('ready', () => {
  conn.exec('pm2 logs bileghurgelt --lines 50 --nostream', (err, stream) => {
    if (err) throw err;
    let output = '';
    stream.on('close', () => {
      console.log('--- LOGS START ---');
      console.log(output);
      console.log('--- LOGS END ---');
      conn.end();
    }).on('data', (data: any) => {
      output += data;
    }).stderr.on('data', (data: any) => {
      output += data;
    });
  });
}).on('error', (err) => {
  console.error('SSH connection error:', err);
}).connect(config);

File: scripts/fetch-logs.py
import paramiko, sys, io
sys.stdout = io.TextIOWrapper(sys.stdout.buffer, encoding='utf-8', errors='replace')
ssh = paramiko.SSHClient()
ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
ssh.connect('13.140.175.47', username='root', password='Aliwdansaa23', timeout=30)
stdin, stdout, stderr = ssh.exec_command('docker logs --tail 20 krono_web', timeout=30)
print(stdout.read().decode('utf-8', errors='replace'))
print(stderr.read().decode('utf-8', errors='replace'))
ssh.close()
